core/src/main/java/org/apache/calcite/rel/core/JoinRelType.java

```diff
@@ -94,6 +142,37 @@ public JoinRelType cancelNullsOnRight() {
       return this;
     }
   }
+
+  /** Transform this JoinRelType to CorrelateJoinType. **/
+  public CorrelateJoinType toLinq4j() {
+    switch (this) {
+    case INNER:
+      return CorrelateJoinType.INNER;
+    case LEFT:
+      return CorrelateJoinType.LEFT;
+    case SEMI:
+      return CorrelateJoinType.SEMI;
+    case ANTI:
+      return CorrelateJoinType.ANTI;
+    }
+    throw new IllegalStateException(
+        "Unable to convert " + this + " to CorrelateJoinType");
+  }
```

Comment:
Do we need this function? Can't we just pass JoinRelType directly?

Reply:
Could we obsolete `CorrelateJoinType` altogether?

Reply:
I think we can, would Deprecate `CorrelateJoinType` in calcite-linq4j

Reply:
Oops, the `calcite-linq4j` module can not see `JoinRelType`, it is a independent module apart from `calcite-core`, actually calcite-core depends on `calcite-linq4j`